tests/pawn-shop.ts

```diff
@@ -527,32 +543,31 @@ describe("PawnHub", () => {
   });
 
   describe("Repay Loan - in SOL", () => {
-    let pawnLoan: any;
-    let pawnTokenAccount: PublicKey;
+    let pawnLoanAddress: PublicKey;
+    let pawnLoanState: any;
 
     beforeEach(async () => {
-      ({ pawnTokenAccount } = await requestLoan(
+      ({ pawnLoan: pawnLoanAddress } = await requestLoan(
         program,
-        pawnLoanKeypair,
+        baseKeypair,
         BORROWER_KEYPAIR,
-        pawnMint.publicKey,
         borrowerPawnTokenAccount,
+        pawnMint.publicKey,
         TERMS_VALID
       ));
 
-      pawnLoan = await program.account.pawnLoan.fetch(
-        pawnLoanKeypair.publicKey
-      );
+      pawnLoanState = await program.account.pawnLoan.fetch(pawnLoanAddress);
 
       await underwriteLoan(
         program,
-        pawnLoan,
-        pawnLoanKeypair,
-        pawnTokenAccount,
+        pawnLoanAddress,
+        pawnLoanState,
         LENDER_KEYPAIR,
         LENDER_KEYPAIR.publicKey,
         BORROWER_KEYPAIR.publicKey
       );
+
+      pawnLoanState = await program.account.pawnLoan.fetch(pawnLoanAddress);
     });
 
     it("Repays loan amount from borrower to lender -- in SOL", async () => {
```

Comment:
should we add a test that it .isfrozen is no longer set?

Reply:
good point, added to seize as well, to ensure the pawn token account is left clean